src/analysis/inner_product_space/lax_milgram.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2022 Daniel Roca González. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Daniel Roca González.
+-/
+import analysis.inner_product_space.projection
+import analysis.inner_product_space.dual
+import analysis.normed_space.banach
+
+/-!
+# The The Lax-Milgram Theorem
+
+
+We consider an Hilbert space `V` over `ℝ`
+equipped with a bilinear form `B : V →L[ℝ] V →L[ℝ] ℝ`.
+For every such form we define the
+`lax_milgram_map : V →L[ℝ] V`,
+which is defined by the property `B (lax_milgram_map B v) w = ⟪v, w⟫`
+using the Fréchet-Riesz representation theorem.
+
+Under the hypothesis that `B` is *coercive* (fulfills `is_coercive`),
+we prove the Lax-Milgram theorem:
+that is, the `lax_milgram_map` can be upgraded to a continuous equivalence `lax_milgram`
+
+## References
+
+TODO cite the slides
+
+
+## Tags
+
+dual, Lax-Milgram
+-/
+
+noncomputable theory
+
+
+
+open_locale classical complex_conjugate
+open is_R_or_C continuous_linear_map linear_map linear_equiv
+
+universe u
+def is_coercive {V : Type u} [inner_product_space ℝ V] [complete_space V] (B : V →L[ℝ] V →L[ℝ] ℝ) : Prop :=
+∃ C, (0 < C) ∧ ∀ u, C * ∥u∥ * ∥u∥ ≤ B u u
+
+
+namespace lax_milgram
+variables {V : Type u} [inner_product_space ℝ V] [complete_space V]
+variables (B : V →L[ℝ] V →L[ℝ] ℝ)
+
+
+def lax_milgram_map (B : V →L[ℝ] V →L[ℝ] ℝ) : V →L[ℝ] V :=
+comp
+  (inner_product_space.to_dual ℝ V).symm.to_continuous_linear_equiv.to_continuous_linear_map
+  B
```

Comment:
This is awkward to write down because of "unification problems".  Here's a slightly more elegant expression, I'll see later whether it can be improved.
```suggestion
↑(id (inner_product_space.to_dual ℝ V).symm.to_continuous_linear_equiv : (dual ℝ V) ≃L[ℝ] V) ∘L
(id B : V →L[ℝ] dual ℝ V)
```

Reply:
Maybe
```suggestion
↑(id (inner_product_space.to_dual ℝ V).symm.to_continuous_linear_equiv : (V →L[ℝ] ℝ) ≃L[ℝ] V) ∘L B
```
is slightly better again.

Reply:
Why the coercion ? What was the problem with the original version? `(inner_product_space.to_dual ℝ V).symm.to_continuous_linear_equiv.to_continuous_linear_map ∘L B` seems to typecheck just fine if we prefer the infix version

Reply:
It can be awkward to have two competing forms of the same thing, like `.to_linear_map` and the coercion.  The API are usually all designed for one of them, and hard to use for the other.  I think, library-wide, we have a preponderance of people thinking of the coercion as the normal form, although I don't think it's a firm policy (yet).

Reply:
(So you can use your way if you prefer, although please use either dot notation `.comp` or infix `∘L` rather than a preceding `comp`.)

Reply:
I see, then I think I will leave your version. What I still don't understand is how the `id` helps with the unification problems...